### PR TITLE
GODRIVER-2002 Remove 'slaveok' from source code and most spec tests

### PR DESCRIPTION
**Pull request by benjirewis**
GODRIVER-2002

Updates a number of spec tests that contain variations of `slaveok` to use variations of `secondaryok`. Removes `SlaveOk` from source code and replaces it with `SecondaryOk`.

The Go driver did not parse a `slaveOK` connection string option, so there was no work there.

### CODE REVIEW

File: mongo/integration/unified/unified_spec_test.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ var (
 		"collection-management",
 		"command-monitoring/unified",
 		"sessions/unified",
+		"retryable-writes/unified",
```
Comment (benjirewis) [inline mongo/integration/unified/unified_spec_test.go:27]: I synced the `retryable-writes` spec tests. This is unrelated to oppressive language removal, but it's always good to update tests as long as they pass. Some retryable writes tests had been ported over to the unified test format, so I added them to the test runner.

**Review by benjirewis — COMMENTED** [review]
I don't believe any of the source code changes are API-breaking since this is all under `x`.

File: mongo/integration/unified_spec_test.go [hunk 2]
```diff
@@ -437,6 +437,19 @@ func executeTestRunnerOperation(mt *mtest.T, testCase *testCase, op *operation,
 		fp, err := op.Arguments.LookupErr("failPoint")
 		assert.Nil(mt, err, "failPoint not found in arguments")
 		mt.SetFailPointFromDocument(fp.Document())
+	case "assertSessionTransactionState":
+		stateVal, err := op.Arguments.LookupErr("state")
+		assert.Nil(mt, err, "state not found in arguments")
+		expectedState, ok := stateVal.StringValueOK()
+		assert.True(mt, ok, "state argument is not a string")
+		actualState := clientSession.TransactionState.String()
```
Comment (kevinAlbs) [inline mongo/integration/unified_spec_test.go:445]: Nit: should this also assert `clientSession` is not `nil`?

**Review by kevinAlbs — COMMENTED** [review]
Changes look good.

I checked with a grep command modified from [this comment](https://jira.mongodb.org/browse/GODRIVER-1862?focusedCommentId=3772604&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-3772604).

```
ggrep -PriI 'slave' --exclude-dir=vendor --exclude-dir=.git .
```

That shows "slave" in the title or contents of SDAM integration tests. Comparing with the specifications repo I think those have been updated. Should those be updated in this PR?

**Review by kevinAlbs — APPROVED** [review]
Per offline discussion, SDAM integration tests will be updated as part of follow-up work. LGTM!

File: mongo/integration/unified_spec_test.go (hunk 2 above)
Comment (benjirewis) [reply mongo/integration/unified_spec_test.go:447]: That's a good idea. I think the only time `clientSession` would be `nil` is if there was a malformed test with an `assertSessionTransactionState` and no corresponding `session` argument, but it's still worth having a check. There are numerous assertions on the fields of `clientSession` in this function without preceding `NotNil` assertions on `clientSession`, so I filed GODRIVER-2117.

**Pull request merged**